packages/amplify-cli/src/extensions/amplify-helpers/resource-status-diff.ts

```diff
@@ -0,0 +1,295 @@
+          }
+    }
+
+    //API :View: Print the resource detail status for the given mutation (Create/Update/Delete)
+    public printResourceDetailStatus = async ( mutationInfo : StackMutationInfo ) => {
+      const header = `${ mutationInfo.consoleStyle(mutationInfo.label)}`;
+      const diff = await this.calculateCfnDiff();
+      print.info(`${resourceDetailSectionStyle(`[\u27A5] Resource Stack: ${capitalize(this.category)}/${this.resourceName}`)} : ${header}`);
+      const diffCount = this.printStackDiff( diff );
+      if ( diffCount === 0 ){
+          console.log("No changes  ")
+      }
+    }
+
+    //API :Data: Calculate the difference in cloudformation templates between local and cloud templates
+    public calculateCfnDiff = async () => {
+      const resourceTemplatePaths = await this.getCfnResourceFilePaths();
+      //load resource template objects
+      this.localTemplate = await this.loadCfnTemplate(resourceTemplatePaths.localTemplatePath)
+      this.cloudTemplate = await this.loadCfnTemplate(resourceTemplatePaths.cloudTemplatePath);
+      const diff : cfnDiff.TemplateDiff = cfnDiff.diffTemplate(this.cloudTemplate, this.localTemplate );
+      return diff;
+    }
+
+    //helper: Select cloudformation file path from build folder or non build folder.
+    private getCfnResourceFilePaths = async() => {
+      return {
+            localTemplatePath : (checkExist(this.resourceFiles.localBuildCfnFile))?this.resourceFiles.localBuildCfnFile: this.resourceFiles.localPreBuildCfnFile ,
+            cloudTemplatePath : (checkExist(this.resourceFiles.cloudBuildCfnFile))?this.resourceFiles.cloudBuildCfnFile: this.resourceFiles.cloudPreBuildCfnFile
+      }
+    }
+
+    //helper: Get category provider from filename
+    private normalizeProviderForFileNames(provider:string):string{
+      //file-names are currently standardized around "cloudformation" not awscloudformation.
+      if ( provider === "awscloudformation"){
+        return CategoryProviders.CLOUDFORMATION
+      } else {
+        return provider;
+      }
+    }
+
+    //helper: Convert cloudformation template diff data using CDK api
+    private printStackDiff = ( templateDiff, stream?: cfnDiff.FormatStream ) =>{
+        // filter out 'AWS::CDK::Metadata' resources from the template
+        if (templateDiff.resources ) {
+          templateDiff.resources = templateDiff.resources.filter(change => {
+            if (!change) { return true; }
+            if (change.newResourceType === 'AWS::CDK::Metadata') { return false; }
+            if (change.oldResourceType === 'AWS::CDK::Metadata') { return false; }
+            return true;
+          });
+        }
+        if (!templateDiff.isEmpty) {
+          cfnDiff.formatDifferences(stream || process.stderr, templateDiff);
+        }
+        return templateDiff.differenceCount;
+    }
+
+    //helper: reads file and deserializes
+    private loadStructuredFile = async(fileName: string, deserializeFunction) => {
```

Comment:
What does this function do? What is the deserialization function here? 
We have a common helper function to read/write files. Can we use that instead?

Reply:
Yes , can refactor to use the common function for reading files and applying deserialization.